### PR TITLE
Feature/UI development

### PR DESCRIPTION
**Pull request by RathodSagar18**

### CODE REVIEW

File: src/components/GeolocationPopup.tsx
Comment (kr396) [inline src/components/GeolocationPopup.tsx:1]: Remove this file

File: src/components/RequestGameCard.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import {createStyleSheet, useStyles} from 'react-native-unistyles';
+import {images} from '../config/images';
+import {Chip} from './Chip';
+import {ThemeButton} from './ThemeButton';
+
+export const RequestGameCard = () => {
+  const {styles, theme} = useStyles(styleSheet);
+  return (
+    <View style={styles.container}>
+      <ScrollView showsVerticalScrollIndicator={false}>
```
Comment (kr396) [inline src/components/RequestGameCard.tsx:12]: Remove scrollview and use flatlist in screen.

File: src/config/images.ts [hunk 2]
```diff
@@ -9,4 +9,5 @@ export const images = {
   homeIcon: require('../assets/images/bottomTabs/home.png'),
   playIcon: require('../assets/images/bottomTabs/play.png'),
   mySportsIcon: require('../assets/images/bottomTabs/mySports.png'),
+  Avtar: require('../assets/images/requestGameCard/avtar.png'),
```
Comment (kr396) [inline src/config/images.ts:12]: A should be small

File: src/navigation/ToptabStack.tsx
Comment (kr396) [inline src/navigation/ToptabStack.tsx:1]: Remove this

File: src/screens/CreateGame/index.tsx [hunk 3]
```diff
@@ -45,7 +45,7 @@ const CreateGame: React.FC<
   const [gameSkillEnabled, setGameSkillEnabled] = useState(false);
   const [skillSliderValues, setSkillSliderValues] = useState<number[]>([1, 5]);
   const [selectedLocation, setSelectedLocation] = useState<string | null>(null);
-  const locationPopup = useRef<BottomSheetModal>(null);
+ const locationPopup = useRef<BottomSheetModal>(null);
```
Comment (kr396) [inline src/screens/CreateGame/index.tsx:48]: Fix spacing

File: src/screens/EditProfile/index.tsx
Comment (kr396) [inline src/screens/EditProfile/index.tsx:1]: Fix formating

File: src/screens/Home/index.tsx [hunk 4]
```diff
@@ -1,12 +1,19 @@
 import {View, Text, ScrollView, SafeAreaView, FlatList} from 'react-native';
-import React from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import {UnistylesRuntime, useStyles} from 'react-native-unistyles';
 import stylesheet from './styles';
 import {GameCard, ThemeButton} from '../../components';
-import {MessageIcon, NotificationIcon} from '../../config/svgs';
+import {ArrowDownIcon, MessageIcon, NotificationIcon} from '../../config/svgs';
+import LocationPopup from '../../components/LocationPopup';
+import {TouchableOpacity} from 'react-native-gesture-handler';
```
Comment (kr396) [inline src/screens/Home/index.tsx:8]: Import this from "react-native"

File: src/screens/Home/index.tsx [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
+import {BottomSheetModal} from '@gorhom/bottom-sheet';
 
-const Home = ({navigation}) => {
+const Home = ({navigation, route}) => {
```
Comment (kr396) [inline src/screens/Home/index.tsx:11]: Is route required here?

File: src/screens/Home/index.tsx [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
   const {styles} = useStyles(stylesheet);
+  const locationPopup = useRef<BottomSheetModal>(null);
```
Comment (kr396) [inline src/screens/Home/index.tsx:13]: Name should be "locationPopupRef"

File: src/screens/Home/index.tsx [hunk 7]
```diff
@@ -1,12 +1,19 @@
...
-import React from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import {UnistylesRuntime, useStyles} from 'react-native-unistyles';
 import stylesheet from './styles';
 import {GameCard, ThemeButton} from '../../components';
-import {MessageIcon, NotificationIcon} from '../../config/svgs';
+import {ArrowDownIcon, MessageIcon, NotificationIcon} from '../../config/svgs';
+import LocationPopup from '../../components/LocationPopup';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {BottomSheetModal} from '@gorhom/bottom-sheet';
```
Comment (kr396) [inline src/screens/Home/index.tsx:9]: BottomSheetModal is this required?

File: src/screens/MySports/index.tsx [hunk 8]
```diff
@@ -6,8 +6,7 @@ import stylesheet from './styles';
 import {format} from 'date-fns';
 
 const MySports = ({navigation, route}) => {
-  const {selectedSport, selectedDate, selectedTime, selectedLocation} =
-    route.params;
+  const {selectedSport, selectedDate, selectedLocation} = route.params || [];
```
Comment (kr396) [inline src/screens/MySports/index.tsx:9]: Remove this

File: src/screens/Play/index.tsx
Comment (kr396) [inline src/screens/Play/index.tsx:1]: Fix formatting issue

**Review by kr396 — APPROVED** [review]

File: src/screens/Home/index.tsx (hunk 7 above)
Comment (RathodSagar18) [reply src/screens/Home/index.tsx:9]: yes,this required to show locationPopupRef you are also used in create game screen

File: android/app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -6,7 +6,10 @@
 
     <uses-feature android:name="android.hardware.camera" android:required="false" />
     <uses-feature android:name="android.hardware.camera.front" android:required="false" />
-    
+
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+
```
Comment (kr396) [inline android/app/src/main/AndroidManifest.xml:11]: Remove extra lines

File: src/screens/Home/index.tsx (hunk 4 above)
Comment (kr396) [reply src/screens/Home/index.tsx:8]: Missed this

**Review by kr396 — CHANGES_REQUESTED** [review]

**Review by kr396 — APPROVED** [review]

**Pull request merged**